### ISSUE TITLE
Proposal: token migration

### ISSUE THREAD

**Issue by Ivshti** [AdExNetwork/adex-protocol-eth#34]
there are various issues with the current implementation of ADX, namely:

- totalSupply is missing
- transfer/transferFrom are not returning `true` as the ERC20 spec requires, breaking every `require(...transfer())` after solidity 0.24: https://medium.com/coinmonks/missing-return-value-bug-at-least-130-tokens-affected-d67bf08521ca

a possible solution might be to migrate the token, which will also give us an opportunity to integrate other token standards (if they're backwards compatible) and re-think some of the structure so that we use less gas

since we don't have an ability to pause the token, the only option is to create the new contract and to allow users to migrate into the new contract

here's an example: https://medium.com/melonport-blog/mln-token-migration-instructions-196a645fc3ac

basically the steps are as follows:

- announce the migration (possibly for mid summer)
- develop and test the contract
- coordinate with exchanges and wallets
- undergo third party audits for the new token contract
- deploy new token contract, migrate on exchanges

**Comment by Ivshti** [comment]
There's no urgent need to do this, plus it's place is not here but in AIPS
